src/main/java/dev/vasyl/proj/controller/BookController.java

```diff
@@ -29,45 +30,50 @@
 @Tag(name = "Book manager",
         description = "API for managing books")
 public class BookController {
-
     private final BookService bookService;
 
+    @PreAuthorize("hasRole('USER') or hasRole('ADMIN')")
     @GetMapping
     @Operation(summary = "Get Book page",
-            description = "Get Book page by page params or/and sort params")
+            description = "Get Book page by page params or/and sort params."
+                    + " Available for User and Admin roles")
     public Page<BookDto> getAll(@ParameterObject
                                     @PageableDefault(size = 20, sort = "title",
                                             direction = Sort.Direction.ASC)
                                             Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
+    @PreAuthorize("hasRole('USER') or hasRole('ADMIN')")
```

Comment:
in your implementation, the user with the role ADMIN has a USER role by default so you can validate only the USER role here